### PR TITLE
docs: Clarify AsyncScript usage

### PR DESCRIPTION
**Pull request by Eideren**
AsyncScript is still a bit too confusing, just clarifying some of its quirks.

### CODE REVIEW

**Comment by Eideren** [comment]
Not too sure how to title/label this one, @VaclavElias feel free to edit anything

File: en/manual/scripts/types-of-script.md [hunk 1]
```diff
@@ -59,14 +59,26 @@ public class SampleAsyncScript : AsyncScript
 	public override async Task Execute() 
 	{
 		// The initialization code should come here, if necessary
+		// This method starts running on the main thread
 		
 		while (Game.IsRunning) // loop until the game ends (optional depending on the script)
 		{
-			await MyEvent;
+			// We're still on the main thread
```
Comment (VaclavElias) [inline en/manual/scripts/types-of-script.md:66]: Shall we still leave `await MyEvent;` below this new comment you added and maybe rename the method? Just to make "clearer"?
```csharp
// We're still on the main thread
await MyMainThreadEvent;
```

Comment (Eideren) [reply en/manual/scripts/types-of-script.md:66]: I don't think we should keep it as `MyEvent`'s purpose was not described, nor was its type/implementation specified. We should have a specific section for it if we were to leave it in I think.

Comment (VaclavElias) [reply en/manual/scripts/types-of-script.md:66]: I see! I was assuming it was just an example how to use it, and where to place the code..

Would this be still a correct implementation (considering your comments), which we could link below your code snippet? I could do a separate PR with the below.

```
Check out an example from our [Async scripts tutorial](https://doc.stride3d.net/latest/en/tutorials/csharpintermediate/async-scripts.html).
```

image

Comment (Eideren) [reply en/manual/scripts/types-of-script.md:66]: Sure, sounds great !

**Comment by VaclavElias** [comment]
Thank you @Eideren

**Pull request merged**